rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,513 @@
+---
+feature: doc-comment-standard
+start-date: 2023-03-27
+author: hsjobeki
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Propose a standard format for Doc-comments.
+
+# Motivation
+[motivation]: #motivation
+
+The following are the envisioned goals
+
+- be able to generate documentation from code for any nix expression.
+- be able to distinguish between documentation-relevant comments and unrelated comments.
+- make doc comments easy to write and read
+- be able to parse and render doc comments nicely
+- standardize a format for doc comments that can be extended by further RFCs
+
+This RFC is a significant change to the existing documentation convention
+but allows to distinguish between regular comments and doc comments. This is important because arbitrary code comments should not end up in generated documentation.
+
+> Hint: Generating static documentation is controvert topic in nixpkgs. It was found that is impossible to generate accurate documentation statically. A correct solution would involve evaluation of expressions in some way. This already goes deeply into implementation details and is thus not further discussed in this document. Although we envision solutions to solve this issue.
```

Comment:
> - it's controversial
> - it was found that it's impossible to generate accurate documentation statically.

I should re-phrase the second sentence (impossible) is quite impossible to prove because there might be a way that is just not found yet. 

References that static generation may be controversial; I don't know if there is a single only source to convince sb.
Also, past the recent comments, I got the feeling that @gilice, @AndersonTorres, and I had a contrary discussion about it.

- Stalled PR Implementation in the repl for doc-comments from @roberth: https://github.com/NixOS/nix/pull/1652
- Stalled PR Implementation from @edolstra: https://github.com/NixOS/nix/pull/5527
- Users asking how to obtain documentation from their code: [discourse thread](https://discourse.nixos.org/t/how-to-generate-documentation-from-arbitrary-nix-code/22292)
- [Any consensus on documentation generation tool for nixpkgs manual?](https://discourse.nixos.org/t/any-consensus-on-documentation-generation-tool-for-nixpkgs-manual/15550)

Reply:
Then this context should be added to the RFC, if you want to keep that sentence in. Otherwise it's a claim without data backing it.